### PR TITLE
#7259: force the current line before probing ahead for the next one

### PR DESCRIPTION
**Pull request by morphqdd**
Fixes the reported case for #7259; see the added `@todo` for the remaining test-coverage part.

`all-lines`'s recursion decided whether to continue by checking `scan.eof` on a freshly constructed `line-or-eof` probe, before the current `scan.line` was ever forced. Since object construction and the `buffer` argument are both lazy, entering the recursion advanced the probe (and its own lookahead byte read) at every level, all the way to the real EOF, before `string buffer` at the base case ever forced any `scan.line`. By then the console stream was already drained, so each line's own multi-byte read got nothing beyond its already-consumed first byte, and every newline came back as an empty line.

Wrapped the recursive step in `seq`, so `scan.line` is forced (consuming exactly that line's own bytes, including its terminator) before the next `line-or-eof` probe is even constructed.

`console.read` reaches the process's real file descriptor 0 through `posix`/`win32`, so an EO `++>` test needs OS-level stdin redirection for the forked test JVM, which this module has none of yet. Left a `@todo` puzzle for that; verified manually against real piped stdin instead:

```
printf 'one\ntwo\nthree\n' | ...  -> "one\ntwo\nthree"
printf 'hello\n'          | ...  -> "hello"
printf 'abc'               | ...  -> "abc"
printf 'one\n\ntwo\n'      | ...  -> "one\n\ntwo"
```

### CODE REVIEW

File: eo-runtime/src/main/eo/stdin.eo [hunk 1]
```diff
@@ -12,6 +12,20 @@
...
 # or just dataize `stdin` object.
+#
+# `all-lines` forces the current line's text through `seq` before the next
+# `line-or-eof` probe is even constructed. Both the accumulated text and
+# that probe read from the same console stream: constructing the probe reads
+# its own lookahead byte, and without this ordering that read races ahead of
+# the current line's own multi-byte read, since neither was otherwise forced
+# until the whole recursion bottomed out at the final `string`.
+#
+# @todo #7259:60min Add `++>` tests exercising `all-lines` and `next-line`
```
Comment (yegor256) [inline eo-runtime/src/main/eo/stdin.eo]: @morphqdd it's technically impossible to mock `posix` or `win32` in EO. the best we can do is use one of our "snippet" tests.

**Comment by morphqdd** [comment]
@yegor256 done, the check is a snippet test now.

The snippet harness could not feed standard input: `EoSourceRun` runs the
program with `exec:java`, which runs inside Maven's own process, and there is
no way to hand that process an input. So a snippet with a `stdin` key is now
forked with `exec:exec` instead, with `<inputFile>` pointing at a file the
harness writes into the fake project. Snippets without the key run exactly as
before.

The new `reads-all-lines` snippet feeds three lines to `stdin.all-lines` and
expects the three lines back. On master it comes back as one byte per line.

I dropped the `@todo` that claimed this could not be tested, and `stdin.eo`
now says why a `++>` test cannot do it and where the real check lives.

**Comment by 0crat** [comment]
@yegor256 Thanks for the review! You've earned +7 points for this: +12 as a basis; -5 for very few (1) comments.  Your running score is [+1916](https://www.eolang.org/zerocracy/objectionary-vitals.html); don't forget to check your [Zerocracy account](https://www.zerocracy.com/) too).

**Comment by 0crat** [comment]
@morphqdd Thanks for the contribution! You've earned +12 points for this: +16 as a basis; -4 for too many hits-of-code (198 >= 100). Please, [keep](https://www.yegor256.com/2018/03/06/speed-vs-quality.html) them coming. Your running score is [+6324](https://www.eolang.org/zerocracy/objectionary-vitals.html); don't forget to check your [Zerocracy account](https://www.zerocracy.com/) too).
